sqlalchemy_dremio/pyodbc.py

```diff
@@ -1,458 +1,108 @@
-
-    def get_schema_names(self, connection, **kw):
-        stmt = select([column('username')],
-               from_obj=[text('dbc.UsersV')],
-               order_by=[text('username')])
-        res = connection.execute(stmt).fetchall()
-        return [self.normalize_name(name['username']) for name in res]
-
-    def get_view_definition(self, connection, view_name, schema=None, **kw):
-
-        if schema is None:
-             schema = self.default_schema_name
-
-        res = connection.execute('show table {}.{}'.format(schema, view_name)).scalar()
-        return self.normalize_name(res)
-
-    def get_view_names(self, connection, schema=None, **kw):
-
-        if schema is None:
-            schema = self.default_schema_name
-
-        stmt = select([column('tablename')],
-                      from_obj=[text('dbc.TablesVX')]).where(
-                      and_(text('DatabaseName = :schema'),
-                           text('tablekind=\'V\'')))
-
-        res = connection.execute(stmt, schema=schema).fetchall()
-        return [self.normalize_name(name['tablename']) for name in res]
+            if (len(_int) - 1) > value.adjusted():
+                result = "%s%s.%s" % (
+                (value < 0 and '-' or ''),
+                "".join(
+                    [str(s) for s in _int][0:value.adjusted() + 1]),
+                "".join(
+                    [str(s) for s in _int][value.adjusted() + 1:]))
+            else:
+                result = "%s%s" % (
+                (value < 0 and '-' or ''),
+                "".join(
+                    [str(s) for s in _int][0:value.adjusted() + 1]))
+        return result
 
-    def get_pk_constraint(self, connection, table_name, schema=None, **kw):
-        """
-        Override
-        TODO: Check if we need PRIMARY Indices or PRIMARY KEY Indices
-        TODO: Check for border cases (No PK Indices)
-        """
 
-        if schema is None:
-            schema = self.default_schema_name
+class DremioExecutionContext_pyodbc(DremioExecutionContext):
+    pass
 
-        stmt = select([column('ColumnName'), column('IndexName')],
-                      from_obj=[text('dbc.Indices')]).where(
-                          and_(text('DatabaseName = :schema'),
-                              text('TableName=:table'),
-                              text('IndexType=:indextype'))
-                      ).order_by(asc(column('IndexNumber')))
 
-        # K for Primary Key
-        res = connection.execute(stmt, schema=schema, table=table_name, indextype='K').fetchall()
+class DremioDialect_pyodbc(PyODBCConnector, DremioDialect):
 
-        index_columns = list()
-        index_name = None
+    execution_ctx_cls = DremioExecutionContext_pyodbc
 
-        for index_column in res:
-            index_columns.append(self.normalize_name(index_column['ColumnName']))
-            index_name = self.normalize_name(index_column['IndexName']) # There should be just one IndexName
+    pyodbc_driver_name = 'Dremio'
```

Comment:
Should be the same as the DRIVER DSN param dependent on platform/OS as specified in base.py